internal/infrastructure/token/encrypt_test.go

```diff
@@ -0,0 +1,34 @@
+package token
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"testing"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestJWKEncryption(t *testing.T) {
```

Comment:
Suggestion (nice to have, not a show stopper) - could we explicitly assert nonce uniqueness?  I think it would suffice to ensure that two encryptions of the same plaintext with same key result in different ciphertext.  (Because the only thing that *could* result in different ciphertext in that scenario is a different nonce.)

Reply:
Sure, done!

I have encrypted the value twice with the same key and verified that the output is different.

Reply:
Cheers mate!